imsim/sky_model.py

```diff
@@ -0,0 +1,113 @@
+        eff_area : `float`
+            Collecting area of telescope in cm^2. Default: Rubin value from
+            https://confluence.lsstcorp.org/display/LKB/LSST+Key+Numbers
+        logger : `logging.Logger`
+            Logger object.
+        """
+        self.exptime = exptime
+        self.mjd = mjd
+        self.eff_area = eff_area
+        self.bandpass = bandpass
+        self.logger = logger
+        self._rubin_sim_sky_model = skybrightness.SkyModel()
+
+    def get_sky_level(self, ra, dec):
+        """
+        Return the sky level in units of photons/arcsec^2 at the
+        specified coordinates.
+
+        Parameters
+        ----------
+        ra : `float`
+            Right Ascension in degrees.
+        dec : `float`
+            Declination in degrees.
+
+        Returns
+        -------
+        `float` : sky level in photons/arcsec^2
+        """
+        # Make a copy of the skybrightness.SkyModel object to avoid
+        # collisions with other threads running this code.
+        rubin_sim_sky_model = copy.deepcopy(self._rubin_sim_sky_model)
+
+        # Set the ra, dec, mjd for the sky SED calculation
+        with warnings.catch_warnings():
+            # Silence astropy IERS warnings.
+            warnings.simplefilter('ignore')
+            rubin_sim_sky_model.setRaDecMjd(ra, dec, self.mjd, degrees=True)
+
+        # Compute the flux in units of photons/cm^2/s/arcsec^2
+        wave, spec = rubin_sim_sky_model.returnWaveSpec()
+        lut = galsim.LookupTable(wave, spec[0])
+        sed = galsim.SED(lut, wave_type='nm', flux_type='flambda')
+        flux = sed.calculateFlux(self.bandpass)
+
+        # Return photons/arcsec^2
+        value = flux * self.eff_area * self.exptime
+
+        if self.logger is not None:
+            self.logger.info("Setting sky level to %.2f photons/arcsec^2 "
+                             "at (ra, dec) = (%s, %s)", value, ra, dec)
+        return value
+
+
+class SkyModelLoader(InputLoader):
+    """
+    Class to load a SkyModel object.
+    """
+    def getKwargs(self, config, base, logger):
+        req = {'exptime': float,
+               'mjd': float}
+        opt = {'eff_area': float}
+        kwargs, safe = galsim.config.GetAllParams(config, base, req=req, opt=opt)
+        kwargs['bandpass'] = base['bandpass']
+        kwargs['logger'] = galsim.config.GetLoggerProxy(logger)
+        return kwargs, safe
+
+
+def sky_level(config, base, value_type):
+    """
+    Use the rubin_sim skybrightness model to return the sky level in
+    photons/arcsec^2 at the center of the current WCS.
+    """
+    sky_model = galsim.config.GetInputObj('sky_model', config, base, 'sky_level')
+
+    kwargs, safe = galsim.config.GetAllParams(config, base)
+
+    wcs = galsim.config.BuildWCS(base['image'], 'wcs', base)
+    value = sky_model.get_sky_level(wcs.center.ra/galsim.degrees,
+                                    wcs.center.dec/galsim.degrees)
```

Comment:
A few things here.
First, `wcs.center` is not necessarily anything we want to use here.  It's a arbitrary location that doesn't have to have any connection to the image.  I think what we want is `base['world_center']`.
Second, I would suggest passing this as a CelestialCoordinate, rather than convert to ra, dec in degrees and then in the other function have LSST sims use degrees.  That way the two uses of degrees, which need to match, are in the same function.
Finally, a slightly simpler syntax to get the angle in degrees is just `ra.deg`.